### PR TITLE
Utils split

### PR DESCRIPTION
**Pull request by Kangz**
PTAL, this moves a bunch of code around to make it possible to share utilities between the samples and the test suite.

### CODE REVIEW

File: src/utils/NXTHelpers.cpp [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "NXTHelpers.h"
+
+#include <shaderc/shaderc.hpp>
+
```
Comment (austinEng) [inline src/utils/NXTHelpers.cpp]: Missing `#include <cstring>` for `strlen`?

File: examples/CMakeLists.txt [hunk 2]
```diff
@@ -12,74 +12,56 @@
...
-
-add_library(utils STATIC ${UTILS_SOURCES})
-target_link_libraries(utils nxt_backend nxt_wire shaderc nxtcpp nxt)
-SetCXX14(utils)
-
-add_executable(CHelloTriangle HelloTriangle.c)
-target_link_libraries(CHelloTriangle utils)
+add_executable(CHelloTriangle CHelloTriangle.cpp)
+target_link_libraries(CHelloTriangle sample_utils)
+SetCXX14(CHelloTriangle)
```
Comment (austinEng) [inline examples/CMakeLists.txt]: Should this still be named `CHelloTriangle `?

File: src/utils/NXTHelpers.cpp (hunk 1 above)
Comment (Kangz) [inline src/utils/NXTHelpers.cpp]: Done

File: examples/CMakeLists.txt (hunk 2 above)
Comment (Kangz) [inline examples/CMakeLists.txt]: It is testing the C API so I think the name is still ok.

**Comment by Kangz** [comment]
Thanks!

**Pull request merged**